### ISSUE TITLE
Improved action buttons for chat code blocks

### ISSUE THREAD

**Issue by toolmantim** [sourcegraph/cody#1096]
Our code block controls have been a problem in recent user testing. Code output in chat is important for anything that still relies on chat, which is most built-in commands, custom commands, and chat itself.

This is a basic building block that we need to get right to instil confidence and lovability. We also lag behind other leading extensions for the manipulating code blocks in chat.

Here's a new design for code block actions that replace our current "Copy" and "Insert at Cursor" buttons that appear on hover:

![image](https://github.com/sourcegraph/cody/assets/153/3104059c-f714-4a3d-9eb6-f4c087566327)

Figma:
https://www.figma.com/file/9112BsKsJc1BpO2j8XYwLL/%F0%9F%A4%96-Cody-VS-Code-%5BMain%5D?type=design&node-id=5110-7877&mode=design&t=KpUpBEhodg72KYQY-4

**Comment by toolmantim** [comment]
I've tightened up the spacing and added the standard quick pick-style dropshadow:
<img src="https://github.com/sourcegraph/cody/assets/153/19be653a-4005-4e88-bc83-4294ad4c17c8" width="376"/>

Also added a prototype to show the experience:
https://www.figma.com/file/9112BsKsJc1BpO2j8XYwLL/%F0%9F%A4%96-Cody-VS-Code-%5BMain%5D?type=design&node-id=5123-9170&mode=design&t=Ey1vVRTX2jpZjtT5-4

https://github.com/sourcegraph/cody/assets/153/43b26a92-bf7f-4274-b46c-573b11d96271

**Comment by yaosamo** [comment]
@toolmantim ah I see it now :) I was talking about the other one tho! 
![CleanShot 2023-09-25 at 17 24 35@2x](https://github.com/sourcegraph/cody/assets/5090118/8749d3ce-557a-45f1-a774-69d9c573abfe)
It's also only visible on hover, maybe that's why I couldn't see it at a glance on first use.

**Comment by toolmantim** [comment]
@yaosamo you found it! Luckily that should be covered by this change too, because the code is shared. But that variant wasn't included in this design because that App chat UI is on pause for the moment.

**Comment by toolmantim** [comment]
@abeatrix looks like we still have some things to clean up before this goes out in the next release, which I'll document here properly soon, but the things I've spotted are:
- Disabled for insert at cursor if no file is open
- Some kind of "Copied" feedback
- Border colors look off on different themes

**Comment by abeatrix** [comment]
@toolmantim do you have a screen shot of the border colors that looks off?

it should be using the vscode color theme for the border:

<img width="589" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/49741e27-82db-49d0-8a5f-8d289288efca">

<img width="628" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/42a7fc72-8eeb-410c-aaa6-548ed3b55f71">

<img width="630" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/d90dfb14-dfa4-42a8-ae6c-d90adb03d0b2">

<img width="626" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/a2eea438-d446-4fec-90a0-9ffc267e1f01">

<img width="640" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/924a247d-f3b8-4ac0-97ee-5683639e47c1">

<img width="641" alt="image" src="https://github.com/sourcegraph/cody/assets/68532117/40bc15ee-b34c-46b5-9d29-79fb2d5be78b">

**Comment by toolmantim** [comment]
@abeatrix yep, here you go! Sometimes the border color is different between the `pre` and the `.container` (themes: Abyss, Kimble Dark, Solarized Dark):

<img width="295" alt="Screenshot 2023-09-26 at 10 33 01 pm" src="https://github.com/sourcegraph/cody/assets/153/9754e0cc-e0c3-4e74-8364-f0b6b0dbd212">
<img width="295" alt="Screenshot 2023-09-26 at 10 32 28 pm" src="https://github.com/sourcegraph/cody/assets/153/148754a3-e722-4f5f-88c7-dce693b7dc6a">
<img width="295" alt="Screenshot 2023-09-26 at 10 32 17 pm" src="https://github.com/sourcegraph/cody/assets/153/870b2d3d-3edc-47b7-bec1-dc8a065f9964">

And often the border is a too bright, so perhaps we're using the wrong variable:

<img width="1074" alt="Screenshot 2023-09-26 at 10 34 40 pm" src="https://github.com/sourcegraph/cody/assets/153/5f3f2ef2-a107-468e-965a-06dd7c3995cc">
<img width="1074" alt="Screenshot 2023-09-26 at 10 34 58 pm" src="https://github.com/sourcegraph/cody/assets/153/aa1e39f4-7994-4a0c-9dd2-5eb12ddc97b7">

**Comment by abeatrix** [comment]
@toolmantim Since the changes for the improved code blocks are in, i've closed this main issue and created a new issue for the bug for the theme border color here: https://github.com/sourcegraph/cody/issues/1268
